### PR TITLE
Convert more paths to std::filesystem::path

### PR DESCRIPTION
**Pull request by hexagonrecursion**

### CODE REVIEW

**Comment by hexagonrecursion** [comment]
An ~advanced~ static analysis technique helped me find many places, where we convert std::string to std::filesystem::path without going through StrUtils::ToString(). Most of this are bugs: std::filesystem::path interprets string in the "native narrow encoding" instead of utf-8.

Here is my technique:
```diff
*** /usr/include/c++/14/bits/fs_path.h.backup   Thu Aug 22 08:31:54 2024
--- /usr/include/c++/14/bits/fs_path.h  Thu Aug 22 09:36:18 2024
***************
*** 321,326 ****
--- 321,330 ----
        _M_cmpts(std::move(__p._M_cmpts))
      { __p.clear(); }
  
+ [[deprecated]]
+     path(const string_type& __source, format = auto_format) {}
+ 
+ [[deprecated]]
      path(string_type&& __source, format = auto_format)
      : _M_pathname(std::move(__source))
      { _M_split_cmpts(); }
***************
*** 358,363 ****
--- 362,369 ----
  
      path& operator=(const path&);
      path& operator=(path&&) noexcept;
+ 
+ [[deprecated]]
      path& operator=(string_type&& __source);
      path& assign(string_type&& __source);
  
***************
*** 380,385 ****
--- 386,394 ----
  
      path& operator/=(const path& __p);
  
+ [[deprecated]]
+     path& operator/=(const string_type& __p) {}
+ 
      template<typename _Source>
        __detail::_Path<_Source>&
        operator/=(_Source const& __source)
***************
*** 407,415 ****
--- 416,428 ----
      // concatenation
  
      path& operator+=(const path& __x);
+ 
+ [[deprecated]]
      path& operator+=(const string_type& __x);
      path& operator+=(const value_type* __x);
      path& operator+=(value_type __x);
+ 
+ [[deprecated]]
      path& operator+=(basic_string_view<value_type> __x);
  
      template<typename _Source>
***************
*** 451,456 ****
--- 464,471 ----
  
      const string_type&  native() const noexcept { return _M_pathname; }
      const value_type*   c_str() const noexcept { return _M_pathname.c_str(); }
+ 
+ [[deprecated]]
      operator string_type() const { return _M_pathname; }
  
      template<typename _CharT, typename _Traits = std::char_traits<_CharT>,
```

File: colobot-base/src/script/scriptfunc.cpp [hunk 1]
```diff
@@ -223,15 +223,11 @@ CBotTypResult CScriptFunctions::cPlayMusic(CBotVar* &var, void* user)
 
 bool CScriptFunctions::rPlayMusic(CBotVar* var, CBotVar* result, int& exception, void* user)
 {
-    std::string filename;
-    std::string cbs;
-    bool repeat;
-
-    cbs = var->GetValString();
-    filename = std::string(cbs);
+    // TODO: I do not know whehter var->GetValString() can return invalid utf-8
```
[the remark this answers is not in the document]
Comment (hexagonrecursion) [reply colobot-base/src/script/scriptfunc.cpp]: Ok. Added try catch. Could you please take a look?

File: colobot-base/src/script/scriptfunc.cpp [hunk 2]
```diff
@@ -223,15 +223,16 @@ CBotTypResult CScriptFunctions::cPlayMusic(CBotVar* &var, void* user)
...
-    std::string cbs;
-    bool repeat;
-
-    cbs = var->GetValString();
-    filename = std::string(cbs);
+    std::filesystem::path filename;
+    try {
+        std::filesystem::path filename = StrUtils::ToPath(var->GetValString());
+    } catch(...) {
+        exception = ERR_UNKNOWN;
```
Comment (melex750) [inline colobot-base/src/script/scriptfunc.cpp:230]: It look like 'filename' variable is re-declared inside the try block and shadows the one before try.

Maybe ```exception = CBotErrFileOpen```("Can't open file") is better feedback than "Internal error - tell the developers".

Comment (hexagonrecursion) [reply colobot-base/src/script/scriptfunc.cpp]: Oops. Thanks. Fixed

**Comment by tomaszkax86** [comment]
I take it you'll replace all the `TempToPath()` calls in future pull requests. It's meant to be used during refactoring to make sure you'll come back and fix the code later.

Looks like there's a merge conflict now. Can you fix it?

**Comment by hexagonrecursion** [comment]
> I take it you'll replace all the `TempToPath()` calls in future pull requests

That was the plan

**Comment by hexagonrecursion** [comment]
> Looks like there's a merge conflict now. Can you fix it?

Ok

**Comment by hexagonrecursion** [comment]
> Looks like there's a merge conflict now. Can you fix it?

Fixed

**Review by tomaszkax86 — APPROVED** [review]

**Pull request merged**